src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java

```diff
@@ -227,7 +228,7 @@ private long getTotalWeight(List<Edge> path, WeightFunction weightFunction) {
         if (path.isEmpty()) {
             return Long.MAX_VALUE;
         }
-        return path.stream().mapToLong(weightFunction::apply).sum();
+        return path.stream().mapToLong(edge -> weightFunction.apply(edge).toLong()).sum();
```

Comment:
nit: I think you can use 
`path.stream().map(weightFunction::apply).map(weightFunction::toLong()).sum()`

Reply:
Fixed.